### DISCUSSION TITLE
Override style manager types.

### DISCUSSION THREAD

**Discussion by skru** [GrapesJS/grapesjs#4454] [General]
I've successfully created a new color input type for the StyleManager using the [docs](https://grapesjs.com/docs/modules/Style-manager.html#customization).

All works fine ([using this](https://github.com/melloware/coloris-npm)) and the added functionality is brilliant (I can change the palette at runtime).

I started to then redefine the style manager properties to then include the new picker. Then realised I can just override the existing by adding the new type with the id as "color":
        
`editor.StyleManager.addType('color', { ... })`

This also works, but the existing color picker is not removed, instead they become stacked and both pickers are useable.

Hidden the old ones with:

`.gjs-field-color {
  display: none !important;
}`

Is this going to cause any issues or is it best to just define a new stylemanager type and redefine all the properties that used the old color picker?

All the best

**Comment by artf** [comment]
Honestly, I'd not expect to see the previous picker, are you able to create a reproducible demo?

**Comment by skru** [comment]
Sorry , should have done that in the first place.

[check it out](https://jsfiddle.net/2vk3z419/17/), notice the default is still usable (have to scroll right)

All the best

**Reply by artf** [reply]
Yeah, right, you should be able to prevent that by adding an empty onRender method.
```js
editor.StyleManager.addType('color', {
  	onRender() {},
        // ....
})
```
